Src/coffeescript/cql-execution/src/datatypes/interval.coffee

```diff
@@ -155,39 +155,45 @@ module.exports.Interval = class Interval
     else
       false
 
-  after: (other) ->
+  after: (other, precision) ->
     closed = @toClosed()
     # Meets spec, but not 100% correct (e.g., (null, 5] after [6, 10] --> null)
     # Simple way to fix it: and w/ not overlaps
     if !!other.toClosed
-      cmp.greaterThan closed.low, other.toClosed().high
+      cmp.greaterThan closed.low, other.toClosed().high, precision
     else
-      cmp.greaterThan closed.low, other
+      cmp.greaterThan closed.low, other, precision
 
-  before: (other) ->
+  before: (other, precision) ->
     closed = @toClosed()
     # Meets spec, but not 100% correct (e.g., (null, 5] after [6, 10] --> null)
     # Simple way to fix it: and w/ not overlaps
     if !!other.toClosed
-      cmp.lessThan closed.high, other.toClosed().low
+      cmp.lessThan closed.high, other.toClosed().low, precision
     else
-      cmp.lessThan closed.high, other
+      cmp.lessThan closed.high, other, precision
 
-  meets: (other) ->
+  meets: (other, precision) ->
     ThreeValuedLogic.or(
-      @meetsBefore(other),
-      @meetsAfter(other)
+      @meetsBefore(other, precision),
+      @meetsAfter(other, precision)
     )
 
-  meetsAfter: (other) ->
+  meetsAfter: (other, precision) ->
     try
-      cmp.equals @toClosed().low, successor(other.toClosed().high)
+      if precision? and @low instanceof DateTime
```

Comment:
Probably a naive question. Why is precision ignored unless the interval is a DateTime?